c7n/resources/sagemaker.py

```diff
@@ -397,13 +397,153 @@ def process(self, resources):
             list(w.map(self.process_instance, resources))
 
 
+@resources.register('sagemaker-model')
+class Model(QueryResourceManager):
+    class resource_type(object):
+        service = 'sagemaker'
+        enum_spec = ('list_models', 'Models', None)
+        detail_spec = (
+            'describe_model', 'ModelName',
+            'ModelName', None)
+        id = 'ModelArn'
+        name = 'ModelName'
+        date = 'CreationTime'
+        dimension = None
+        filter_name = None
+
+    filter_registry = filters
+    permissions = ('sagemaker:ListTags',)
+
+    def augment(self, resources):
+        def _augment(r):
+            client = local_session(self.session_factory).client('sagemaker')
+            tags = client.list_tags(
+                ResourceArn=r['ModelArn'])['Tags']
+            r.setdefault('Tags', []).extend(tags)
+            return r
+
+        with self.executor_factory(max_workers=1) as w:
+            return list(filter(None, w.map(_augment, resources)))
+
+
+@Model.action_registry.register('delete')
+class DeleteModel(BaseAction, StateTransitionFilter):
+    """Deletes sagemaker-model(s)
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+          - name: delete-sagemaker-model
+            resource: sagemaker-model
+            filters:
+              - "tag:DeleteMe": present
+            actions:
+              - delete
+    """
+    schema = type_schema('delete')
+    permissions = ('sagemaker:DeleteModel',)
+
+    def process_instance(self, resource):
+        client = local_session(
+            self.manager.session_factory).client('sagemaker')
+        try:
+            client.delete_model(
+                ModelName=resource['ModelName'])
+        except ClientError as e:
+            self.log.exception(
```

Comment:
fwiw, best practice is to catch known errors, and raise otherwise. i know that the original implementer didn't handle that well, but the most common catchable on delete is resource not found, we generally want to know on other states, to ensure proper modeling of a service's exceptions. logged errors are typically not caught by most users (minus perhaps those using tools/c7n_sentry).